### PR TITLE
dev test labs artifacts and virtual machine modules

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
New modules for DTL

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
azure_rm_devtestlabartifactsource
azure_rm_devtestlabvirtualmachine

### CODE REVIEW

**Comment by zikalino** [comment]
@mattclay is there a way to get any GitHub token from the env?
if now could we add one?
for this test it just needs to be valid GitHub access token.

**Comment by mattclay** [comment]
@zikalino We don't have one currently. What does the token need access to?

**Comment by zikalino** [comment]
@mattclay the funny thing is it doesn't have to have any special access to anywhere ;-)
In the test I am using artifacts from public GitHub repo.... but still the DTL API requires valid GitHub token.
So any token will do, preferably stored in GITHUB_TOKEN env variable.

**Comment by mattclay** [comment]
@zikalino It seems we could create a GitHub account and then issue a read-only token with no privileges for that account and use it for the test. However, I have a few concerns with that:

- That usage may violate the [GitHub Terms of Service](https://help.github.com/articles/github-terms-of-service/#3-account-requirements).
- The token would be public, which may leave it open to abuse, even with no privileges.
- The token would need to be obfuscated to avoid detection by [Token Scanning](https://help.github.com/articles/about-token-scanning/).

So although it's technically feasible, it's probably not a good idea to do so.

**Comment by zikalino** [comment]
@mattclay so perhaps I will just disable this test for now....
Indeed, if token is stored via Shippable, it won't be detected by Token Scanning, but could be potentially extracted by somebody running a test... maybe we could discuss more tomorrow.

File: test/integration/targets/azure_rm_devtestlab/tasks/main.yml [hunk 1]
```diff
@@ -96,6 +99,160 @@
     that:
       - output.changed
 
+#- name: Create instance of DevTest Labs artifacts source
```
Comment (yungezz) [inline test/integration/targets/azure_rm_devtestlab/tasks/main.yml:176]: why tests are commented out?

Comment (zikalino) [reply test/integration/targets/azure_rm_devtestlab/tasks/main.yml:176]: no GitHub token in the CI, and  @mattclay had concerns about adding one, so we agreed to disable this particular test. Please check discussion above.

**Comment by nitzmahone** [comment]
@zikalino if you want to fix the merge conflict, I'll merge this

**Comment by zikalino** [comment]
@nitzmahone done! let's wait for the tests to complete

**Comment by zikalino** [comment]
@nitzmahone now merged and tests completed

**Pull request merged**